sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -105,6 +105,9 @@ case class DataSource(
       case _ => cls
     }
   }
+
+  private def providingInstance(): Any = providingClass.getConstructor().newInstance()
```

Comment:
It's definitely at least an `AnyRef`. Can you write `private def [T <: AnyRef] providingInstance(): T`?  You can even tighten the bound with a cast, as we know the supertype of what's being returned.
However I don't think it's worth it. You can use AnyRef or omit the type here.

Reply:
Yes ,I think it's not worth too. I omit return type here.